File: src/main/java/de/lulonaut/Bot/commands/messagecount/DeleteAllMessages.java
package de.lulonaut.Bot.commands.messagecount;

import de.lulonaut.Bot.commands.Aliases;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;

import java.util.EnumSet;
import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class DeleteAllMessages extends ListenerAdapter {
    List<String> aliases = Stream.of(Aliases.DeleteMessagesAliases.values())
            .map(Enum::name)
            .collect(Collectors.toList());


    @Override
    public void onGuildMessageReceived(GuildMessageReceivedEvent event) {
        String[] msg = event.getMessage().getContentRaw().split(" ");
        if (event.getAuthor().isBot()) {
            return;
        }
        if (!aliases.contains(msg[0].substring(1))) {
            return;
        }

        EmbedBuilder eb = new EmbedBuilder();
        EnumSet<Permission> perms = Objects.requireNonNull(event.getMember()).getPermissions();
        if (!perms.contains(Permission.MANAGE_SERVER)) {
            eb.setTitle("Error!");
            eb.setDescription("");
            event.getChannel().sendMessage(eb.build()).queue();
            return;
        }
        //user has perms
        event.getChannel().sendMessage("Please type \"confirm\" to reset the counter for this server. **THIS CANNOT BE UNDONE**").queue();
        DeletingStateMachine dsm = new DeletingStateMachine(event.getChannel(), event.getMember().getUser());
        event.getJDA().addEventListener(dsm);
    }
}

File: src/main/kotlin/de/lulonaut/bot/commands/messagecount/DeleteAllMessages.kt
package de.lulonaut.bot.commands.messagecount

import de.lulonaut.bot.cache.DatabaseCache
import de.lulonaut.bot.utils.Conf
import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter
import java.util.*

class DeleteAllMessages : ListenerAdapter() {
    override fun onGuildMessageReceived(event: GuildMessageReceivedEvent) {
        val msg = event.message.contentRaw.split(" ").toTypedArray()
        if (event.author.isBot) {
            return
        }
        val prefix = DatabaseCache.getConfig(event.guild.id)?.get("prefix")

        if (!msg[0].equals(prefix + "deleteMessages", ignoreCase = true)) {
            return
        }

        val eb = EmbedBuilder()
        val perms = Objects.requireNonNull(event.member)!!.permissions
        if (!perms.contains(Permission.MANAGE_SERVER)) {
            eb.setTitle("Error!")
            eb.setDescription("You don't have the `Manage Server` permission.")
            event.channel.sendMessage(eb.build()).queue()
            return
        }
        //user has perms
        event.channel.sendMessage("Please type \"confirm\" to reset the counter for this server. **THIS CANNOT BE UNDONE**")
            .queue()
        val dsm = DeletingStateMachine(event.channel, event.member!!.user)
        event.jda.addEventListener(dsm)
    }
}